runtime/lua/vim/lsp/util.lua

```diff
@@ -1632,6 +1632,7 @@ function M.open_floating_preview(contents, syntax, opts)
           vim.b[bufnr].lsp_floating_preview = nil
         end
         api.nvim_del_augroup_by_name(augroup_name)
+        return true
```

Comment:
Why is the augroup needed if we are returning `true` here? I guess to avoid "duplicate" WinClosed events? But at least, `nvim_del_augroup_by_name` shouldn't be necessary because of the `if not ok` check on line 1626.

Should we instead create a single, global `WinClosed` handler that checks the win-id against `vim.b[bufnr].lsp_floating_preview` ?

Reply:
>  I guess to avoid "duplicate" WinClosed events?

yes because there can reuse a floating-window `  local floating_winnr = opts._update_win` in line 1531. 

Does the global WinClosed mean that winid is not needed as pattern? Any window closing will trigger it. Another thing we are missing is a function to check the existence of augroup or autocmd? The current method of checking the existence of augroup always uses pcall wrapper. Checking whether it has been registered an event requires `#nvim_get_autocmd({filter}) > 0`. It would be better if the filter supports `count` then return count number or something else 🤔

Reply:
Yes. exists can check everything on vim side. It would be better if we have a simple lua api to match it on autocmd level .  not a big problem.  nvm